scripts/languageserver/provider_misc.jl

```diff
@@ -21,7 +22,9 @@ end
 function process(r::Request{Val{Symbol("textDocument/didOpen")},DidOpenTextDocumentParams}, server)
     server.documents[r.params.textDocument.uri] = split(r.params.textDocument.text,r"\r\n?|\n")
     
-    process_diagnostics(r.params.textDocument.uri, server)
+    if r.params.textDocument.uri[1:min(length(r.params.textDocument.uri),length(server.rootPath))] == server.rootPath 
```

Comment:
I'm wondering what the right behavior is when ``rootPath`` is ``""``... We probably still want to lint in that case? Maybe we actually need to rules: 1) if a folder is open, only lint files from that folder, 2) if no folder is open, lint any file, except files from the base library? I'm not sure, though...